pkg/packet/bgp/bgp.go

```diff
@@ -12325,6 +12360,48 @@ func NewPathAttributeUnknown(flags BGPAttrFlag, typ BGPAttrType, value []byte) *
 	}
 }
 
+// BGPUpdateAttributes defines a map with a key as bgp attribute type
+// and value as bool. Value set to true indicates that the attribute specified by the key
+// exists in the bgp update.
+type BGPUpdateAttributes struct {
+	Attribute map[BGPAttrType]bool
+}
+
+func GetBGPUpdateAttributes(data []byte) BGPUpdateAttributes {
+	m := BGPUpdateAttributes{
+		Attribute: make(map[BGPAttrType]bool),
+	}
+	for p := 0; p < len(data); {
+		flag := data[p]
+		p++
+		t := data[p]
+		m.Attribute[BGPAttrType(t)] = true
+		p++
+		var l uint16
+		// Checking for Extened
+		if flag&0x10 == 0x10 {
+			l = binary.BigEndian.Uint16(data[p : p+2])
```

Comment:
this is safe? data contains 2 bytes at least?

Reply:
good catch, adding safety check.